source/opt/folding_rules.cpp

```diff
@@ -1430,6 +1430,64 @@ FoldingRule FactorAddMuls() {
   };
 }
 
+// Replaces |inst| inplace with an FMA instruction |(x*y)+a|.
+void ReplaceWithFma(Instruction* inst, uint32_t x, uint32_t y, uint32_t a) {
+  uint32_t ext =
+      inst->context()->get_feature_mgr()->GetExtInstImportId_GLSLstd450();
+
+  if (ext == 0) {
+    inst->context()->AddExtInstImport("GLSL.std.450");
+    ext = inst->context()->get_feature_mgr()->GetExtInstImportId_GLSLstd450();
+    assert(ext != 0 &&
+           "Could not add the GLSL.std.450 extended instruction set");
```

Comment:
Is the caller responsible for checking that std.450 is available, or is that implied by `IsFloatingPointFoldingAllowed()`? I'm just wondering if this should be a soft or hard error.

Reply:
The previous line 1439, should add it, so this should never fail.  I'm trying to check if in the future it cannot be added because of something that is changed in the Vulkan spec.  I feel the hard error is appropriate.